### PR TITLE
Added support for alternate pid file location

### PR DESCRIPTION
**Pull request by k0ekk0ek**
I've added support for writing the pid file to an alternate location using a /pidfile=PATH command line switch because it's customary to store the pid file in /var/run/<proc>.pid whereas the other files Sonarr writes are better placed in /var/lib (or /var/cache depending on their purpose). The switch is only visible on non-Windows platforms.

### CODE REVIEW

File: src/NzbDrone.Common/Processes/PidFileProvider.cs [hunk 1]
```diff
@@ -12,12 +12,14 @@ public interface IProvidePidFile
 
     public class PidFileProvider : IProvidePidFile
     {
+        private readonly IStartupContext _startupContext;
         private readonly IAppFolderInfo _appFolderInfo;
         private readonly IProcessProvider _processProvider;
         private readonly Logger _logger;
 
-        public PidFileProvider(IAppFolderInfo appFolderInfo, IProcessProvider processProvider, Logger logger)
+        public PidFileProvider(IStartupContext startupContext, IAppFolderInfo appFolderInfo, IProcessProvider processProvider, Logger logger)
```
Comment (markus101) [inline src/NzbDrone.Common/Processes/PidFileProvider.cs]: A bit pedantic, but the new class should be put at the end of the list (but before logger).

```
public PidFileProvider(IAppFolderInfo appFolderInfo, IProcessProvider processProvider, IStartupContext startupContext, Logger logger)
```

**Comment by markus101** [comment]
Thanks for the PR.

One issue with this (not specific to your changes), is the switches are lost during a restart of Sonarr from the UI. Specific to your changes, you'll want to add this switch to the PreservedArguments in StartupContext: https://github.com/Sonarr/Sonarr/blob/develop/src/NzbDrone.Common/EnvironmentInfo/StartupContext.cs#L67

**Comment by k0ekk0ek** [comment]
Processed the comments and added the /pidfile flag to the list of preserved arguments.

**Comment by Taloth** [comment]
@markus101 Won't the pidpath be lost if restarted via NzbDrone.Update?

**Comment by k0ekk0ek** [comment]
Support for this should be implemented in the configuration file.

**Pull request closed without merging**